File: problem14/Solution.java
public class Solution {
	public static void main(String[] args) {
		int nums[] = {1,12,-5,-6,50,3}, k = 4;
		System.out.println(findMaxAverage(nums, k));
	}
	public static double findMaxAverage(int[] nums, int k) {
		int max = 0;
		for (int i = 0; i < k; i++)
			max += nums[i];
        int curr = max;
		for (int i = k; i < nums.length; i++) {
			curr = curr - nums[i - k] + nums[i];
			max = Math.max(curr, max);
		}
		return (double) max / k;
    }
}

File: problem14/Solution.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        maxAvg = sum(nums[len(nums)-k:])
        print(f"maxAvg = {maxAvg}")
        l, r = len(nums)-k, len(nums) - 1
        tmp = maxAvg
        while l > 0:
            l -= 1
            tmp = tmp + nums[l] - nums[r]
            r -= 1
            maxAvg = max(maxAvg, tmp)
        print(maxAvg)
        return maxAvg / k
